lib/matplotlib/ticker.py

```diff
@@ -1212,12 +1216,32 @@ class EngFormatter(Formatter):
          24: "Y"
     }
 
-    def __init__(self, unit="", places=None):
+    def __init__(self, unit="", places=None, sep=" "):
+        """
+        Parameters
+        ----------
+        unit: string (default: "")
+            Unit symbol to use.
+
+        places: int (default: None)
+            Precision, i.e. number of digits after the decimal point.
+            If it is None, falls back to the floating point format '%g'.
+
+        sep: string (default: " ")
+            String used between the value and the prefix/unit. Beside the
+            default behavior, some other useful use cases may be:
+            * sep="" to append directly the prefix/unit to the value;
+            * sep="\\u00a0" to use a no-break space;
```

Comment:
Why is a no-break space useful here?  You have to go way out of your way to get mpl to try to wrap your text.